src/sys/unix/tcp/listener.rs

```diff
@@ -13,6 +13,12 @@ pub struct TcpListener {
 }
 
 impl TcpListener {
+    pub(crate) fn from_std(inner: net::TcpListener) -> Self {
+        let raw_fd = inner.into_raw_fd();
```

Comment:
We are you used `into_raw` and then `from_raw` here? You just return `TcpStream { inner }`.